kiegroup/kogito-cloud-operator
# [KOGITO-1294] Adding support for binary builds

See:
https://issues.redhat.com/browse/KOGITO-1294

@radtriste @DuncanDoyle this is a preliminary version for the Binary Builds support. I still have some work to do:

- [x] Review why Build Configurations are triggering too much reconciliation requests
- [x] Introduce Kogito CLI `kogito create-service <my-kogito-service>`

To test it, you will need:

1) Update the KogitoApp CRD on your cluster with the one from this PR
2) Install the operator from this image: `quay.io/ricardozanini/kogito-cloud-operator:0.9.0-rc1` (use the yaml files from `deploy\`)
3) Use this example to create your own app:
```yaml
apiVersion: app.kiegroup.org/v1alpha1
kind: KogitoApp
metadata:
  name: example-quarkus-flash
spec:
  build:
    imageVersion: 0.8.0-rc3
    imageRuntimeTag: quay.io/ricardozanini/kogito-quarkus-jvm-ubi8:0.9.0-rc1
```
4) Compile any project and upload it to the cluster with:
```shell
oc start-build example-quarkus-flash-binary --from-dir=target
```
Tomorrow I will work in your feedback and remaining tasks to turn this into a green PR.

Thank you!

Many thanks for submiting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.MD#sending-a-pull-request)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Your feature/bug fix has a unit test that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster
